ManeuverQueue_KSP1.2/ManeuverQueue.cs

```diff
@@ -83,143 +93,239 @@ protected List<Vessel> defaultVessels
+						else
+						{
+							if (sb == null) sb = new System.Text.StringBuilder();
+							sb.Length = 0;
+							sb.Append("ManeuverQueue: VesselIconSprite.");
+							foreach (char c in f.Name)
+							{
+								if (c != '_' && !char.IsLetterOrDigit(c))
+									sb.AppendFormat("\\x{0:X}", (uint)c);
+								else sb.Append(c);
+							}
+							sb.AppendFormat("({0}): {1}", f.Name.Length, f.FieldType.FullName);
+							Debug.Log(sb.ToString());
+							if ((vesselImageField == null || f.Name == "\x1")
+								&& f.FieldType == typeof(Image))
+								vesselImageField = f;
+						}
+					}
+				}
 			}
 
-			this.pluginConfiguration.SetValue(ManeuverQueue.configurationFiltersKey, (int)MapViewFiltering.VesselTypeFilter.All);
-			this.pluginConfiguration.save();
+			pluginConfiguration.load();
 
-			this.spaceTrackingScene = (SpaceTracking)UnityEngine.Object.FindObjectOfType(typeof(SpaceTracking));
+			savedFilterState =
+				MapViewFiltering.vesselTypeFilter == MapViewFiltering.VesselTypeFilter.All
+				? (MapViewFiltering.VesselTypeFilter)pluginConfiguration
+					.GetValue(configurationFiltersKey,
+					(int)MapViewFiltering.VesselTypeFilter.All)
+				: MapViewFiltering.vesselTypeFilter;
 
-			this.sideBarRect = GetSideBarRect();
+			pluginConfiguration.SetValue(configurationFiltersKey,
+				(int)MapViewFiltering.VesselTypeFilter.All);
+			pluginConfiguration.save();
 
-			this.windowPos = new Rect(this.sideBarRect.xMax, WINDOW_VERTICAL_POSITION, 10, 10);
+			sideBarRect = GetSideBarRect();
 
-			this.windowStyle = new GUIStyle(HighLogic.Skin.window)
+			windowPos = new Rect(sideBarRect.xMax, WINDOW_VERTICAL_POSITION, 10, 10);
+
+			windowStyle = new GUIStyle(HighLogic.Skin.window)
 			{
 				margin = new RectOffset(),
 				padding = new RectOffset(5, 5, 5, 5)
 			};
 
-			GameEvents.onGameSceneSwitchRequested.Add(this.onGameSceneSwitchRequested);
-
-			GameEvents.onVesselDestroy.Add(this.onVesselDestroy);
-			GameEvents.onVesselCreate.Add(this.onVesselCreate);
-			GameEvents.onKnowledgeChanged.Add(this.onKnowledgeChanged);
-			GameEvents.OnMapViewFiltersModified.Add(this.onMapViewFiltersModified);
+			GameEvents.onGameSceneSwitchRequested.Add(onGameSceneSwitchRequested);
 
-			ManeuverQueue.filterModeLabels = Enum.GetValues(typeof(FilterMode)).Cast<FilterMode>().Where(
-				x => x != FilterMode.Undefined).Select(
-					x => ManeuverQueue.LabelForFilterMode(x)).ToArray();
+			GameEvents.onVesselDestroy.Add(onVesselDestroy);
+			GameEvents.onVesselCreate.Add(onVesselCreate);
+			GameEvents.onKnowledgeChanged.Add(onKnowledgeChanged);
+			GameEvents.OnMapViewFiltersModified.Add(onMapViewFiltersModified);
 
-			this.currentMode = (FilterMode)this.pluginConfiguration.GetValue(ManeuverQueue.configurationModeKey, (int)FilterMode.Default);
+			if (DefaultVessels != null)
+			{
+				delaySetMode = false;
+				CurrentMode = (FilterMode)pluginConfiguration.GetValue(configurationModeKey, (int)FilterMode.Default);
+			}
+			else
+			{

```

Comment:
Possibly always use delaySetMode (or StartCoroutine).
Can Unity call SpaceTracking.Start() and ManeuverQueue.Start() in parallel? That would be total mess.